packages/route/src/Infrastructure/SymfonyCmf/Routing/CmfRouteProvider.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Route\Infrastructure\SymfonyCmf\Routing;
+
+use Sulu\Route\Application\Routing\RouteCollectionForRequestLoaderInterface;
+use Symfony\Cmf\Component\Routing\RouteProviderInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Exception\RouteNotFoundException;
+use Symfony\Component\Routing\Route as SymfonyRoute;
+use Symfony\Component\Routing\RouteCollection;
+
+/**
+ * @final
+ *
+ * @internal No BC promises are given for this class. It may be changed or removed at any time.
+ */
+class CmfRouteProvider implements RouteProviderInterface
+{
+    /**
+     * @param RouteCollectionForRequestLoaderInterface[] $routeCollectionForRequestLoaders
+     */
+    public function __construct(
+        private iterable $routeCollectionForRequestLoaders,
+    ) {
+    }
```

Comment:
The whole idea is that this is our only connection point with the `Symfony\Cmf`. Marked it as `@internal` as this shouldn't be changed or used by projects. Also we are even free the replace it if in the future dynamic router is maybe part of symfony core or a new major version of it is released with changes.

With our own `RouteCollectionForRequestLoaderInterface` which just has `public function getRouteCollectionForRequest(Request $request): RouteCollection;` we have all about routing then in our own control.

I also added the `RouteDefaultsProviderInterface` which is very similar to our existing RouteBundle entity of 2.6 version. But simplified it that every resourceKey requires an own defaults provider.